server/src/main/java/org/opensearch/action/admin/cluster/shards/CatShardsResponse.java

```diff
@@ -23,28 +28,44 @@
  */
 public class CatShardsResponse extends ActionResponse {
 
-    private ClusterStateResponse clusterStateResponse = null;
-
-    private IndicesStatsResponse indicesStatsResponse = null;
+    private IndicesStatsResponse indicesStatsResponse;
+    private DiscoveryNodes nodes = DiscoveryNodes.EMPTY_NODES;
```

Comment:
We have removed ClusterStateResponse with DiscoveryNodes, as the API was only using Nodes from ClusterState. Since we have now started to send ```List<ShardRouting> responseShards``` as well separately, we decided to avoid sending full clusterState over the transport.
Conversation [Link](https://github.com/opensearch-project/OpenSearch/pull/14641#discussion_r1780739771).